library/spdm_common_lib/libspdm_com_crypto_service.c

```diff
@@ -175,6 +175,20 @@ static bool libspdm_calculate_m1m2(void *context, bool is_mut,
     hash_size = libspdm_get_hash_size(
         spdm_context->connection_info.algorithm.base_hash_algo);
 
+    LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO, "message_a data :\n"));
+    libspdm_internal_dump_hex(
+        libspdm_get_managed_buffer(&spdm_context->transcript.message_a),
+        libspdm_get_managed_buffer_size(
+            &spdm_context->transcript.message_a));
+    status = libspdm_append_managed_buffer(
+        m1m2,
+        libspdm_get_managed_buffer(&spdm_context->transcript.message_a),
+        libspdm_get_managed_buffer_size(
+            &spdm_context->transcript.message_a));
+    if (LIBSPDM_STATUS_IS_ERROR(status)) {
+        return false;
+    }
```

Comment:
I think we need add SPDM version check and only add mut auth for SPDM 1.2 and above.